document/src/vespa/document/select/resultlist.cpp

```diff
@@ -96,39 +97,56 @@ ResultList::combineVariables(
 ResultList
 ResultList::operator&&(const ResultList& other) const
 {
-    ResultList result;
+    ResultList results;
 
     // TODO: optimize
+    vespalib::hash_set<uint32_t> resultForNoVariables;
```

Comment:
Since we're dealing with a fixed set of possible values, this could also be a bit set to avoid added overhead with allocation per operator invocation. Same goes for the other places we do this.

Reply:
Fixed